### DISCUSSION TITLE
Project Settings

### DISCUSSION THREAD

**Discussion by hifabienne** [zitadel/zitadel#2263] [Ideas]
We already have some settings you are able to set on a project.
Do we need some more settings on the project?

Current Settings:
- Assert Roles on Authentication
- Check roles on Authentication

Coming soon:
- Check has project on Authentication
- Private Labeling Setting (Which private labeling poicy should trigger on the project, the organisation of the project or  the user?)

Ideas of settings:
- Organisation in which the user should be registered
- Login Policy Setting: A setting do define which login policy should trigger on default. At the moment it is always the ZITADEL Login Policy. With the primary domain scope it is possible to trigger your organisation, but then you are only able to login with a user of your organisation.
- Check is User from my organisation

What do you think about it and do you have some more ideas?

**Comment by mffap** [comment]
The settings should also solve the following cases in my opinion. 

Given that my company ACME develops MySaaS. 
- When a B2C user want to login, then she should be able enter loginname / use social login at the MySaaS login page and authenticate for MySaaS.
- When a user from a business partner (foobar.com) wants to login, and the organisation foobar.com exists in ZITADEL, and the organisation foobar.com has configured login via it's own IDP (eg, Azure AD), then the user is being redirected from MySaaS login page to the configured IDP for authentication at MySaaS.

**Reply by fschmied** [reply]
That would be very useful.

> and the organisation foobar.com exists in ZITADEL

... and there is an organization grant from MySaaS to foobar.com, right? That way, ACME has some level of control about this feature.

> and the organisation foobar.com has configured login via it's own IDP (eg, Azure AD), then the user is being redirected from MySaaS login page to the configured IDP for authentication at MySaaS

It would probably be a good idea to have this depend on an "auto-redirect" setting within the foobar.com organization. So foobar.com can decide whether they want to present a username/password login in addition to the IdP or not.

So, in effect, the MySaaS login would detect whether the username given is from a business partner (i.e., one of the granted organizations) or not. If it's from a business partner, e.g., foobar.com, it would either show the business partner's login page or automatically redirect to that business partner's IdP if the granted organization says to do so. That would be very useful, yet versatile.

**Reply by hifabienne** [reply]
I think this would be a great feature, but it's not a project setting.
This should be configured somewhere in the login policy or on the external identity providers.

**Comment by mffap** [comment]
We should enable a use case where an organization owns multiple products with different private labeling. Each product would be represented as a project. 

Example: CAOS Ltd. is the organization and has two products ZITADEL and ORBOS. When I login into ORBOS (theoretically), then I see the ORBOS Logo and Colors. When I login into ZITADEL, then I see the ZITADEl logo, colors, text etc.
